internal/streamingnode/server/wal/interceptors/segment/manager/partition_manager.go

```diff
@@ -72,17 +78,20 @@ func (m *partitionSegmentManager) AssignSegment(ctx context.Context, req *Assign
 	// So it's just a promise check here.
 	// If the request time tick is less than the fenced time tick, the assign operation is fenced.
 	// A special error will be returned to indicate the assign operation is fenced.
-	// The wal will retry it with new timetick.
 	if req.TimeTick <= m.fencedAssignTimeTick {
 		return nil, ErrFencedAssign
 	}
 	return m.assignSegment(ctx, req)
 }
 
-// SealAllSegmentsAndFenceUntil seals all segments and fence assign until the maximum of timetick or max time tick.
-func (m *partitionSegmentManager) SealAllSegmentsAndFenceUntil(timeTick uint64) (sealedSegments []*segmentAllocManager) {
+// SealAndFenceSegmentUntil seal all segment that contains the message less than the incoming timetick.
+func (m *partitionSegmentManager) SealAndFenceSegmentUntil(timeTick uint64) (sealedSegments []*segmentAllocManager) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
+	// no-op if the incoming time tick is less than the fenced time tick.
+	if timeTick <= m.fencedAssignTimeTick {
```

Comment:
When would this happen?

Reply:
1. Concurrent maunalflush incoming, The older manualflush will be ignored.
2. If manualflush trigger a flushsegment operation, it will redo the operation to generate a new timetick to keep the message sequence：flushTs -> flushsegment -> flushsegment -> manualflush. So the redo operation will be rejected here. In previous implementation, the sequence will be flushTs -> manualflush -> flushsegment -> flushsegment, it's wierd.